app/migration_v_0_7.go

```diff
@@ -0,0 +1,19 @@
+package app
+
+import (
+	"crypto/sha256"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (app *App) Migration07(ctx sdk.Context) {
+	all_resources := app.resourceKeeper.GetAllResources(&ctx)
```

Comment:
Loading everything into memory is not the best approach. Resources can be large. I would suggest to use iterator instead and load resources one by one.

Reply:
Left a todo in code